release.md

```diff
@@ -0,0 +1,298 @@
+
+The biggest change is that `truffle-contract` now returns numbers as [BN](https://github.com/indutny/bn.js/) by default instead of BigNumber. This is necessary - BigNumber doesn't have the precision required by the EVM. The number libraries share some methods (notably `.toNumber()`) but their arithmetic operations are quite different.
+
+**Important**: If you receive a number directly from web3 (e.g. *not* from a truffle-contract instance)  you'll get a **string**. Luckily there's a helpful BN conversion method in the web3 utils.
+```javascript
+const stringBalance = await web3.eth.getBalance('0xabc..');
+const bnBalance = web3.utils.toBN(stringBalance);
+```
+
+**Helpful Resources**
++  Web3 1.0's documentation. Well worth taking a look - lots of differences in their new API.
++ EthWork's [bn-chai](https://github.com/EthWorks/bn-chai) library: BN helpers for your tests. (💡 ProTip courtesy elenadimitrova)
+
+------------
+### ⚠️ Breaking Changes ⚠️
+
+| Category | V4 (Web3 0.0) |  V5 (Web3 1.0) |
+| ------ | ---- | ------- |
+| addresses (return value) | lower-case | check-summed (mixed-case) |
+| numbers  (return value)  | BigNumber | BN (configurable) |
+| tuples (return value)       | Array | Object w/ named & indexed keys |
+| `.contract` (underlying Web3 abstraction)| same as now | completely [different](https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html) |
+|`.at` (TruffleContract method)| sync / then-able | async |
+
+-----------
+### 🍨 Features 🍨
+
+### Methods / `.new` have an EventEmitter interface *and*  return a promise.
+```javascript
+example
+  .setValue(123)
+  .on('transactionHash', hash => {} )
+  .on('receipt', receipt => {})
+  .on('error', error => {})
+  .on('confirmation', (num, receipt) => {} )
+  .then( receipt => {} )
+```
+
+### Events have an EventEmitter interface
+```javascript
+example
+  .ExampleEvent()
+  .on('data', event => ... etc ... )
+
+example
+  .ExampleEvent()
+  .once('data', event => ... etc ... )
+```
+### Reason strings!! Find out the reason.
+```javascript
+# In a Solidity file
+require(msg.sender == owner, 'not authorized');
+
+// In JS
+try {
+  await example.transferToSelf({from: nonOwner})
+} catch (err) {
+  assert(err.reason === 'not authorized');
+  assert(err.message.includes('not authorized');
+}
+```
+
+### Overloaded Solidity methods (credit to @rudolfix and @mcdee)
+```javascript
+example.methods['setValue(uint256)'](123);
+example.methods['setValue(uint256,uint256)'](11,55);
+```
+
+### Configure format of number return values
+As mentioned above - `truffle-contract` now returns BN. We've made this configurable so if you have an existing test suite you'd prefer to gradually transition from BigNumber to BN, it's possible to configure a contract's number format as below.
```

Comment:
`BigNumber.js` etc.

Maybe these should be links; or maybe we should just make sure everything gets linked at least once.

Reply:
Done